argschema/utils.py

```diff
@@ -252,31 +331,16 @@ def build_schema_arguments(schema, arguments=None, path=None, description =None)
                 if isinstance(validator,mm.validate.OneOf):
                     arg['help']+= " (valid options are {})".format(validator.choices)
 
-            field_type = type(field)
-            if isinstance(field, mm.fields.Boolean):
-                arg['type'] = ast.literal_eval
-            #catch lists to figure out desired type
-            elif isinstance(field, mm.fields.List):
+            if (isinstance(field, mm.fields.List) and
+                not field.metadata.get("cli_as_single_argument", False)):
+                warnings.warn(("'{}' is using old-style command-line syntax "
```

Comment:
I think the warning should provide better instructions about how to change their code to adopt the future format (i.e. add cli_as_single_argument=True) and provide an example (--mylist 1 2 3 4 to --mlist [1,2,3,4])